### PR TITLE
Add Markdown validation / style checking

### PR DESCRIPTION
**Pull request by marians**
This PR adds markdown linting (automatic checking of rules) of the `src` folder content as a mandatory step in CI. As a necessary evil, the rules are applied to all markdown files.

See [this example](https://app.circleci.com/pipelines/github/giantswarm/docs/551/workflows/2cd1c3c2-ca68-4afe-b203-a25bec07eb49/jobs/2749/steps) to understand what happens when linting fails.

For extra 🍬 this also works locally with VS code and the `davidanson.vscode-markdownlint` add-on ([Link](https://github.com/DavidAnson/vscode-markdownlint)). Example:

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,6 +56,14 @@ build: vendor build-css
 		quay.io/giantswarm/crd-docs-generator:0.0.0-cce07b5b225bd1b7ad576a0e1f5ebb773397915c \
 		  --config /opt/crd-docs-generator/config/crd-docs-generator-config.yaml
 
+lint:
+	docker run \
+	  -v ${PWD}:/workdir \
+	  -w /workdir \
+	  06kellyjac/markdownlint-cli:latest \
```
Comment (axbarsan) [inline Makefile:63]: Should we use a fixed version number, so we won't have breaking changes when they upgrade?

File: src/content/guides/creating-clusters-via-crs-on-aws/index.md [hunk 2]
```diff
@@ -12,23 +12,23 @@ This guide will show you how to create tenant clusters by creating and applying
...
 
 All the tenant clusters, created with release version 10.x.x+, are managed as [Cluster API](https://github.com/kubernetes-sigs/cluster-api) [custom resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) in the Control Plane.
 The Cluster API is a Kubernetes project to bring declarative, Kubernetes-style APIs to cluster creation, configuration, and management. It provides optional, additive functionality on top of core Kubernetes.
 
 At a high-level, the Cluster API is used to manage two types of CRs:
-  - [Cluster](/reference/cp-k8s-api/clusters.cluster.x-k8s.io/) - represents a Kubernetes control plane.
-  - [MachineDeployment](/reference/cp-k8s-api/machinedeployments.cluster.x-k8s.io/) -  represents a node pool.
+
+- [Cluster](/reference/cp-k8s-api/clusters.cluster.x-k8s.io/) - represents a Kubernetes control plane.
+- [MachineDeployment](/reference/cp-k8s-api/machinedeployments.cluster.x-k8s.io/) -  represents a node pool.
```
Comment (axbarsan) [inline src/content/guides/creating-clusters-via-crs-on-aws/index.md:26]:
```suggestion
- [MachineDeployment](/reference/cp-k8s-api/machinedeployments.cluster.x-k8s.io/) - represents a node pool.
```

File: src/content/reference/gsctl/create-kubeconfig.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ The output of the command gives details on what exactly happens.
 The next example shows creation of a self-contained configuration file:
 
 ```nohighlight
-$ gsctl create kubeconfig -c w6wn8 --self-contained kubeconfig.yaml
+gsctl create kubeconfig -c w6wn8 --self-contained kubeconfig.yaml
```
Comment (axbarsan) [inline src/content/reference/gsctl/create-kubeconfig.md:65]:
```suggestion
gsctl create kubeconfig --cluster w6wn8 --self-contained kubeconfig.yaml
```

File: Makefile (hunk 1 above)
Comment (marians) [inline Makefile:63]:
```suggestion
	  06kellyjac/markdownlint-cli:0:21:0 \
```

Use specific docker image tag

File: Makefile [hunk 4]
```diff
@@ -56,6 +56,14 @@ build: vendor build-css
 		quay.io/giantswarm/crd-docs-generator:0.0.0-cce07b5b225bd1b7ad576a0e1f5ebb773397915c \
 		  --config /opt/crd-docs-generator/config/crd-docs-generator-config.yaml
 
+lint:
+	docker run \
+	  -v ${PWD}:/workdir \
+	  -w /workdir \
+	  06kellyjac/markdownlint-cli:0:21:0 \
```
Comment (marians) [inline Makefile:63]:
```suggestion
	  06kellyjac/markdownlint-cli:0.21.0 \
```

**Pull request merged**